### ISSUE TITLE
[Feature Request]: Option to crop windows (to hide the titlebar)

### ISSUE THREAD

**Issue by lucas75** [pop-os/cosmic-epoch#940]
I was following some issues asking to hide the title bar of all applications and as I understood this is not possible because many windows draw their own decorations. Therefore I have an alternative proposal that may help users to achieve the same workflow and I was wondering is it is possible.

The idea is outline in this sketch:

![image](https://github.com/user-attachments/assets/408f09cc-d22c-4939-bd23-1663bc64c387)
IMG1: This image shows a user clicking with the right mouse button in a application window while pressing the super-key which displays the menu with options to "create window stack", "minimize", "maximize". In this menu there is a new option "crop" and the user selects it.

![image](https://github.com/user-attachments/assets/0beb4664-dc83-4b6c-bcf0-d9b6963fa18f)
This image shows the proposed configure window dialog that present the group "crop margins of this window class" with fields for top, bottom, left and right margins. The user place 30 in the top crop input and confirms it. 

![image](https://github.com/user-attachments/assets/c29388df-f5c1-4a38-ab84-a478e315a470)
This image shows the proposed result. The user sees the same window without the titlebar. The windows continues to draw itself with the titlebar, but the desktop environment simple crop the margin according to the configuration.

This is possible?

**Comment by ealtun21** [comment]
A settings in the settings panel or something inside the config file to disable the title-bar would be amazing for me.

**Comment by lucas75** [comment]
> A settings in the settings panel or something inside the config file to disable the title-bar would be amazing for me.

Yes, this would be great. But as I understood of other conversations, most apps draw their own title-bars which prevents the desktop environment to know the the title-bar size and/or if the title-bar have custom menus or buttons. Gnome extensions to hide the title bar does seem to have the same issue because they doesn't work for many apps. The crop window option is for those apps. 

I don't know the internals of the desktop environment, but I suspect the necessary infrastructure should already be there, after all, animations does far more complicated things with the window content than cropping.

**Comment by vinibortoletto** [comment]
I'd really like this feature too. Title bars don't have much use when using tiling mode. 

On Gnome PopOS, there's an option in the tiling extension to hide title bars, why wouldn't it be possible to do the same in Cosmic? I think window manager like i3wm also remove title bars by default.

**Comment by NaharEmet** [comment]
this is required +++
or at least let us shrink the title bar. when tilling terminal windows the title bar takes up meaningful percentages of screen real estate to provide no value. the tiling management with hotkeys works very well. i don't know what kind of animal would use the title-bar to drag windows around.
